tests/unit/test_data_utils.py

```diff
@@ -0,0 +1,19 @@
+import os
+import unittest2 as unittest
+import numpy as np
+from hdmf.container import Data
+
+from hdmf.backends.hdf5 import H5DataIO
+
+class TestDataIO(unittest.TestCase):
+
+    def test_set_dataio(self):
+        """
+        Test that Data.set_dataio works as intended
+        """
+        dataio = H5DataIO(chunks=(1, 1, 3))
+        data = np.arange(30).reshape(5, 2, 3)
+        container = Data('wrapped_data', data)
+        container.set_dataio(dataio)
+        self.assertIs(dataio.data, data)
+        self.assertIs(dataio, container.data)
```

Comment:
I think this test should be added here https://github.com/hdmf-dev/hdmf/blob/dev/tests/unit/utils_test/test_core_DataIO.py to make sure we don't have DataIO tests in multiple places.